.github/workflows/test.yml

```diff
@@ -139,7 +139,10 @@ jobs:
       - name: go test race
         if: ${{ !startsWith(matrix.go, '1.18.') && !startsWith(matrix.go, '1.19.') }}
         run: |
-          go test -race -shuffle=on -v -count=10 ./...
+          if [[ "$(uname)" == "Darwin" ]]; then
+            env CGO_ENABLED=0 go test -race -shuffle=on -v -count=10 ./...
```

Comment:
Why is this working only with Darwin?

Reply:
Only macOS allows building with the race detector and Cgo disabled. All other platforms fail to build with the message: `go: -race requires cgo`

Reply:
I didn't know that. Actually Cgo was required even on Windows.

Let's separate the tasks like this

```yaml
      - name: go test race (no Cgo)
        if: ${{ runner.os == 'macOS' &&  !startsWith(matrix.go, '1.18.') && !startsWith(matrix.go, '1.19.') }}
        run: |
          # -race requires Cgo except for macOS. See https://go.dev/doc/articles/race_detector#Requirements
          env CGO_ENABLED=0 go test -race -shuffle=on -v -count=10 ./...
      - name: go test race (Cgo)
        if: ${{ !startsWith(matrix.go, '1.18.') && !startsWith(matrix.go, '1.19.') }}
        run: |
          env CGO_ENABLED=1 go test -race -shuffle=on -v -count=10 ./...
```